lib/iris/tests/unit/util/test_mask_cube_from_shapefile.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import pytest
+import shapely
+
+from iris.coords import DimCoord
+import iris.cube
+from iris.exceptions import IrisUserWarning
+import iris.tests as tests
+from iris.util import mask_cube_from_shapefile
+
+
+class TestBasicCubeMasking(tests.IrisTest):
+    """Unit tests for mask_cube_from_shapefile function"""
+
+    def setUp(self):
+        basic_data = np.array([[1, 2, 3], [4, 8, 12]])
+        self.basic_cube = iris.cube.Cube(basic_data)
+        coord = DimCoord(
+            np.array([0, 1.0]),
+            standard_name="projection_y_coordinate",
+            bounds=[[0, 0.5], [0.5, 1]],
+            units="1",
+        )
+        self.basic_cube.add_dim_coord(coord, 0)
+        coord = DimCoord(
+            np.array([0, 1.0, 1.5]),
+            standard_name="projection_x_coordinate",
+            bounds=[[0, 0.5], [0.5, 1], [1, 1.5]],
+            units="1",
+        )
+        self.basic_cube.add_dim_coord(coord, 1)
+
+    def test_basic_cube_intersect(self):
+        shape = shapely.geometry.box(0.6, 0.6, 0.9, 0.9)
+        masked_cube = mask_cube_from_shapefile(shape, self.basic_cube)
+        assert (
+            np.sum(masked_cube.data) == 8
+        ), f"basic cube masking failed test - expected 8 got {np.sum(masked_cube.data)}"
+
```

Comment:
While it is clear that we **_do_** work wth higher-dimensional input cubes, 
since there is an integration test with `["NetCDF", "global", "xyt", "SMALL_hires_wind_u_for_ipcc4.nc"]`, 
I think it would be useful to have to testcases for operation over additional dimensions, and to make it clear.
E.G. Does this rely on numpy broadcasting, therefore depending on dimension ordering (rather than coordinate axis matching) ?

To be clear, I really don't think we need to be very clever here.  But we _should_ document + test what usage is supported.

Reply:
it does just rely on numpy broadcasting (I think from mask_cube anyways) - I'm not too sure how that works myself. If you can point to 4d or non-ordered dimension data to test on I can see how it performs.

Reply:
Added integration test against 4D file in iris-test-data